### DISCUSSION TITLE
Auto-indent not working in some circumstances

### DISCUSSION THREAD

**Discussion by pati08** [NvChad/NvChad#2513] [Q&A] [answered]
I'm programming for WASM frontend in the Yew.rs framework, and I'm experiencing an issue with auto-indentation from nvim. Inside of the `html!` macro, a macro that essentially allows for writing preprocessed HTML directly in rust (a bit like JSX), no matter how the auto-indent should normally act, all newlines are started at the left without indentation. Why is this? Can I do some config magic to fix it?

For example, I by default get this:
```rs
html!(
<div>
<p>{ "hello, world" }</p>
</div>
)
```

instead of:
```rs
html!(
    <div>
      <p>{ "hello, world" }</p>
    </div>
)
```

**Comment by siduck** [comment]
does it work without nvchad? i hope you have html tsparser installed

**Comment by h0pes** [comment, marked as answer]
> I'm programming for WASM frontend in the Yew.rs framework, and I'm experiencing an issue with auto-indentation from nvim. Inside of the `html!` macro, a macro that essentially allows for writing preprocessed HTML directly in rust (a bit like JSX), no matter how the auto-indent should normally act, all newlines are started at the left without indentation. Why is this? Can I do some config magic to fix it?
> 
> For example, I by default get this:
> 
> ```rust
> html!(
> <div>
> <p>{ "hello, world" }</p>
> </div>
> )
> ```
> 
> instead of:
> 
> ```rust
> html!(
>     <div>
>       <p>{ "hello, world" }</p>
>     </div>
> )
> ```
@nvim-ftw do you have Yew `html!` macro expansion with formatting and autocompletion working in NvChad?
